src/main/java/io/cryostat/rules/MatchExpressionValidator.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The Cryostat Authors
+ *
+ * The Universal Permissive License (UPL), Version 1.0
+ *
+ * Subject to the condition set forth below, permission is hereby granted to any
+ * person obtaining a copy of this software, associated documentation and/or data
+ * (collectively the "Software"), free of charge and under any and all copyright
+ * rights in the Software, and any and all patent rights owned or freely
+ * licensable by each licensor hereunder covering either (i) the unmodified
+ * Software as contributed to or provided by such licensor, or (ii) the Larger
+ * Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.rules;
+
+import jdk.nashorn.api.tree.CompilationUnitTree;
+import jdk.nashorn.api.tree.Parser;
+import jdk.nashorn.api.tree.TreeVisitor;
+
+public class MatchExpressionValidator {
+
+    private final Parser parser = Parser.create();
+
+    private final TreeVisitor<Void, String> treeVisitor = new MatchExpressionTreeVisitor();
+
+    String validate(Rule rule) {
```

Comment:
This is for my own understanding of Java, but why doesn't the `validate` method need to state `throws Exception? The reason I'm asking is that in the corresponding unit test, either an `IllegalMatchExpressionException` or a `NashornException` is possible when calling this method.

Reply:
`IllegalMatchExpressionException extends RuntimeException`. Java has a distinction between checked and unchecked exceptions. Checked exceptions must be declared thrown and explicitly caught somewhere, but unchecked exceptions don't. NullPointerException is another example of an unchecked exception you've probably seen before - it would really, really suck if NPEs always had to be accounted for with try/catch blocks or throws declarations, because then nearly every line of Java code would need exception handling.

Reply:
Ah makes sense, thanks.

Reply:
I would say in this case that it probably would make sense for the exception we throw to be a checked kind, since the `TreeVisitor` interface doesn't declare that the methods throw checked exceptions, I can't implement the interface and have mine change the method signatures by adding a new checked exception declaration.

I suppose I could work around this by having the `TreeVisitor` throw an unchecked type, then have the `validator` catch and rethrow those as checked exceptions. That's probably a good idea, actually.